android/pytorch_android/src/main/cpp/pytorch_jni_common.cpp

```diff
@@ -10,6 +10,54 @@
 
 namespace pytorch_jni {
 
+bool Trace::is_initialized_ = false;
+
+#if defined(TRACE_ENABLED) && defined(__ANDROID__)
+Trace::fp_ATrace_beginSection Trace::ATrace_beginSection = {};
+Trace::fp_ATrace_endSection Trace::ATrace_endSection = {};
+#endif
+
+void Trace::ensureInit() {
```

Comment:
Consider moving all of these function definitions (except for init) into the class definition in the header so they can be inlined.